src/context/UserContext.js

```diff
@@ -0,0 +1,19 @@
+import { createContext, useContext, useState } from 'react';
+
+const UserContext = createContext();
+
+export function UserProvider({ children }) {
+  const [user, setUser] = useState('');
+
+  const contextValue = { user, setUser };
+
+  return <UserContext.Provider value={contextValue}>{children}</UserContext.Provider>;
+}
+
+export function useUser() {
+  const context = useContext(UserContext);
+  if (context === undefined) {
+    throw new Error('Error: ...');
```

Comment:
Let's be more descriptive in the error message. If we're throwing it when context is undefined, we can include that in the message.